From 48a0b8b0f726b02d57b9f45282b6c20532cfd959 Mon Sep 17 00:00:00 2001
From: Pawel Moll <pawel.moll@st.com>
Date: Fri, 17 Oct 2008 11:56:05 +0100
Subject: [PATCH] sound_stm: Non-BPA2 buffer memory management fixed

Only the BPA2-sourced buffer is explicitly ioremaped, but
iounmap was called always. Fixed now.

Signed-off-by: Pawel Moll <pawel.moll@st.com>
---
 sound/stm/common.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/stm/common.c b/sound/stm/common.c
index fad6bd9..5fa12dc 100644
--- a/sound/stm/common.c
+++ b/sound/stm/common.c
@@ -468,10 +468,10 @@ void snd_stm_buffer_free(struct snd_stm_buffer *buffer)
 			"dma_bytes=%u\n", runtime->dma_addr,
 			runtime->dma_area, runtime->dma_bytes);
 
-	iounmap(runtime->dma_area);
-
 	if (buffer->bpa2_part) {
 #if defined(CONFIG_BPA2)
+		iounmap(runtime->dma_area);
+
 		bpa2_free_pages(buffer->bpa2_part, runtime->dma_addr);
 		runtime->dma_area = NULL;
 		runtime->dma_addr = 0;
-- 
1.5.3.6

